### ISSUE TITLE
CSS in style attributes

### ISSUE THREAD

**Issue by KamasamaK** [textmate/html.tmbundle#88]
Would it cause any issues to add `source.css` to the value of `style` attributes?

**Comment by infininight** [comment]
Currently this isn't really possible due to how the CSS grammar is structured. The HTML grammar is already structured to be ready for when the CSS grammar is restructured/redone though.

**Comment by revelt** [comment]
hi all,
Is there are ticket issued for _CSS grammar to be re-structured_?

**If so**, please link here.
**If not**, let's create one (I'm not sure where exactly) so people can see it's got a chance of being worked on.

Currently, looking at this ticket, I don't see what's "the next step", is the issue moving. It seems VSCode's HTML inline CSS highlighting depends on this issue.

**Comment by jamilabreu** [comment]
@infininight i hope this is the right place to ask! I'm looking into this issue, any guidance at all would be appreciated:

https://github.com/dracula/visual-studio-code/issues/86#issuecomment-495434222

Image: https://user-images.githubusercontent.com/33581079/44486178-3baf1500-a629-11e8-80c6-5734c1858faa.png

**Comment by josheanes** [comment]
As HTML email developers, we rely heavily on inline CSS syntax highlighting. VS not having this is the only reason I haven't switched over. Here's a screenshot illustrating the difference in having CSS inline syntax highlighting or not - https://i.imgur.com/h5EkNXE.png

Does anyone think this will ever be fixed/implemented?

**Comment by artyil** [comment]
Is it going to be supported? What is the implementation complexity here? I can try to help

**Comment by sky47sky** [comment]
Can this be looked into? This is one of the last things I'm trying to customize after switching to VSCode.

**Comment by Sahaya-Noel** [comment]
Yeah Please try to implement this. This is the only feature that makes me switch to sublime text and back to VSCode. Thanks in advance!!

**Comment by josheanes** [comment]
Anyone know of any workarounds?

**Comment by MarcosBaungartner** [comment]
Syntax highlighting for inline CSS in HTML is very important! It would be great if VSCode developers could implement this feature.

**Comment by derek-lowery** [comment]
+1

**Comment by nxcco** [comment]
Is this still active or has it been forgotten?

**Comment by EthanA120** [comment]
So much time has passed since the issue has been posted and still nothing happened, nothing yet?

**Comment by justintroy** [comment]
+1

**Comment by mark3apps** [comment]
+1

**Comment by ItsCubeTime** [comment]
+1


Theres an extension that enables this btw: https://marketplace.visualstudio.com/items?itemName=outofsync42.inline-css-color (albeit its a little tedious to setup as it requires you to manually modify your settings.json)

**Comment by predaytor** [comment]
+1

**Comment by CiberNin** [comment]
I'm certain that interest for this feature remains strong.

**Comment by rusproject** [comment]
+1. Take my vote for this feature. This is especially an issue when trying to switch from Sublime to VSCode to refactor older projects with a lot of inline CSS.

**Comment by gnat** [comment]
Kinda wild this ain't in gents.
